calico_cni_k8s_test.go

```diff
@@ -396,6 +396,95 @@ var _ = Describe("CalicoCni", func() {
 					}))
 				})
 			})
+
+			Context("Using host-local IPAM: request an IP then release it, and then request it again", func() {
+				It("Should successfully assign IP both times and successfully release it in the middle", func() {
+					netconfHostLocalIPAM := fmt.Sprintf(`
+				  {
+					"name": "net5",
+					  "type": "calico",
+					  "etcd_endpoints": "http://%s:2379",
+					  "ipam": {
+					    "type": "host-local",
+						"subnet": "usePodCidr"
+					  },
+				   "kubernetes": {
+				     "k8s_api_root": "http://127.0.0.1:8080"
+			    	},
+			   	  "policy": {"type": "k8s"},
+				  "log_level":"info"
+					}`, os.Getenv("ETCD_IP"))
+
+					config, err := clientcmd.DefaultClientConfig.ClientConfig()
+					Expect(err).NotTo(HaveOccurred())
+
+					clientset, err := kubernetes.NewForConfig(config)
+					Expect(err).NotTo(HaveOccurred())
+
+					// Create a K8s Node object with PodCIDR and name equal to hostname.
+					_, err = clientset.Nodes().Create(&v1.Node{
+						ObjectMeta: v1.ObjectMeta{Name: hostname},
+						Spec: v1.NodeSpec{
+							PodCIDR: "10.0.0.0/24",
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// Step 1: Create the first pod on node we created.
+					name := fmt.Sprintf("run%d", rand.Uint32())
+
+					// Create a K8s pod with the node we created above.
+					_, err = clientset.Pods(K8S_TEST_NS).Create(&v1.Pod{
+						ObjectMeta: v1.ObjectMeta{Name: name},
+						Spec: v1.PodSpec{Containers: []v1.Container{{
+							Name:  fmt.Sprintf("container-%s", name),
+							Image: "ignore",
+						}},
+							NodeName: hostname,
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// host-local IPAM always assigns the IPs sequentialls except the first 2 that are reserved
```

Comment:
I'm not a huge fan of relying on this internal understanding of how the host-local IPAM plugin assigns IP addresses.

This could easily change in the future and would break this test.  Can we find another way of asserting that the plugin released the IP address?